internal/image/image.go

```diff
@@ -30,61 +31,112 @@ type ScanResults struct {
 }
 
 type Image struct {
-	flattenedLayers []fileMap
-	innerImage      *v1.Image
-	extractDir      string
+	// Final layer is the last element in the slice
+	layers         []imgLayer
+	innerImage     *v1.Image
+	extractDir     string
+	baseImageIndex int
+	configFile     *v1.ConfigFile
+	layerIDToIndex map[string]int
 }
 
-func (img *Image) LastLayer() fileMap {
-	return img.flattenedLayers[len(img.flattenedLayers)-1]
+func (img *Image) LayerIDToIdx(id string) int {
+	return img.layerIDToIndex[id]
+}
+
+// LayerIDToCommand takes in a layer id and returns the history CreatedBy field
+// of the corresponding layer
+func (img *Image) LayerIDToCommand(id string) string {
+	file, err := (*img.innerImage).ConfigFile()
+	if err != nil {
+		log.Panicln("Failed to get inner image config file")
+	}
+	idxCount := img.layerIDToIndex[id]
+	var i int
+	// Match history to layer IDX by skipping empty layer history entries
+	for i = 0; idxCount >= 0; i++ {
```

Comment:
why is this `>= 0` as opposed to `> 0` ? 

If `idxCount == 0`, that means we've found `img.layerIDToIndex[id]` non-empty layers right? at that point `file.History[i]` should contain what we want?

Reply:
I believe `>=` is correct. because the layer index is 0 based. Imagine a simple image with just 1 non-empty layer:
```
  | 0
  | 1
0 | 2
  | 3

i = 0
idxCount = 0

i = 1
idxCount = 0

i = 2
idxCount = 0

i = 3
idxCount = -1
```

so the return layer is (`i - 1 = 2`).

Not sure what's the best way to clarify this in the comments, should I put the example above in?

Reply:
Ah my bad, yeah `idxCount` is 0-indexed in the first place. Ignore this comment then!